### DISCUSSION TITLE
Tapo App 3.8.103.. Android...

### DISCUSSION THREAD

**Discussion by home409ca** [JurajNyiri/HomeAssistant-Tapo-Control#748] [Discuss]
Hi,  
A new version of the HACS add-on has been released. I immediately connected the camera to the internet, updated the software, and when I wanted to change the settings in the Tapo App, I noticed that I don’t have version 3.8.103 but an older one...  

Currently, I don’t have access to the camera via Home Assistant...  

Does anyone have a link to download the latest Android version with the option to enable "Third-Party Compatibility"?  

Thanks!

**Comment by talormanda** [comment]
I don't see it live yet on the play store for myself. Hoping it will come in later today.

**Comment by ssersh** [comment]
Just now only available in IOS App store...

**Comment by Happy-Cadaver** [comment]
At least one person has had success getting the beta and making it work. However I toggled on beta, got added, but still have no update. YMMV

**Reply by robvanoostenrijk** [reply]
I've been on the beta for the last year and the version available for me is 3.7.811 only. So definitely not all beta users are getting the 3.8.103 version yet.

**Comment by Bobbik1** [comment]
If anyone has beta 3.8  , please share the APK

**Comment by galvani** [comment]
I did get an iphone from a friend and enabled it. Works fine

**Reply by TeiruzuX** [reply]
I also activated it with someone else's iPhone, but it's been 2 hours and it doesn't work

**Comment by Heisenberg2018** [comment]
With the update, can an Android user upload the APK once so that we can all install it?

**Comment by costinr1** [comment]
With IOS it's ok but how to use motion detection event? Using smartthings in HA, there is a  tapo motion sensor that can be triggered, but in this HACS local device "cell motion detection" is not available ("This entity is no longer being provided by the tapo_control integration") I'm using tapo c200 and c210.

**Reply by Maas79** [reply]
I find the Integration starts to report this when the detection is turned off on the App (or there has been no alert detected). Make sure the alert is on and that there has been detections and reinitialise the device in HA.

**Reply by costinr1** [reply]
Thanks, you are right, there must be some movement in order to be correctly set-up (the app detection option was already set), but I think this is a bug, the sensor must remain active in this HACS implementation, as long as the option in the initial config panel is checked.

**Reply by JurajNyiri** [reply]
Not a bug, read Binary sensor for motion doesn't show up or work under FAQ in Readme.

**Reply by costinr1** [reply]
> Not a bug, read Binary sensor for motion doesn't show up or work under FAQ in Readme.

Thank you very much Juraj for your effort!! (I'm waiting this implementation since it stopped working a long time ago), just my opinion, I'm not sure I understand why is this behavior of the motion sensor detection, as long as the option is checked in the first config screen, but it's working ok, thank you.

**Reply by JurajNyiri** [reply]
When that option is enabled integration asks the camera to report whatever motion it can. Then camera tells integration when it sees something - could be pet, person, motion… integration has no way of knowing what the camera can and will report and will create an entity for each event category when the camera reports it.

**Reply by JurajNyiri** [reply]
This works the same way as the official HA onvif component and is a standard.

**Comment by Heisenberg2018** [comment]
Can someone who has already activated the new app check whether the privacy mode for the cameras can now be turned on and off again?

**Reply by Maas79** [reply]
Yes can be controlled by HA

**Comment by siankatabg** [comment]
Someone uploaded the app to apkmirror, I've just installed it and got the new switch enabled.

**Reply by home409ca** [reply]
thanks!

**Reply by Heisenberg2018** [reply]
After a year everything is finally back to normal, what can the activation of a single switch in an app do, here in iobroker the adapter finally starts again without any errors! 👌

**Reply by Nico83500** [reply]
Thanks !

**Reply by Heisenberg2018** [reply]
Rejoiced too soon, the adapter starts without errors in iobroker, the socket works again, I can see some values of the camera, but I can't send remote commands to the camera, e.g. privacy mode on or off.

**Comment by ddonevski** [comment]
I just got the new app from Play Store.

**Reply by Nico83500** [reply]
Me too from Play Store France

**Comment by asanchezed** [comment]
I have dowloaded the apk from here [https://www.apkmirror.com/apk/tp-link-global-inc/tp-link-tapo/tp-link-tapo-3-8-113-release/](url) and now I have the cameras working in home assistant again!

**Comment by ivan11paths** [comment]
Has anyone received the update on Play Store Spain?

**Reply by JurajNyiri** [reply]
APK is available in readme.

**Comment by AJ-Reckless** [comment]
I've updated the app (Android) and enabled the third party support switch but I still get an invalid cloud password error.  I gave it about 10 hours after flipping the switch in case the adjustment needed time and even reset my password.  Still no luck.  Anyone else experiencing anything similar?

**Reply by JurajNyiri** [reply]
Follow the instructions in FAQ.

**Reply by asanchezed** [reply]
Open the app and select the camera so that the app will update the camera data with the information that appears in the lab section. Then you will be able to authenticate again in home assistant.

**Reply by Naesstrom** [reply]
I have the same problem to with a few cameras, most worked once I opened the camera. Will give it until tomorrow!

**Reply by AJ-Reckless** [reply]
I was able to get things connected again.  In case anyone else has the same issue and followed all of the steps in the FAQ I had to disable third party support and re-enable it again in order to get the system to accept my cloud password.  While I can log in now, I have a separate issue where I'm getting prompted multiple times a day to reauthenticate for each of my cameras.  The integration says my password(s) changed (they haven't).  I just resubmit the same credentials and things work again for a short period of time before I have to do it all over again.  At least I'm moving in the right direction, haha.

**Comment by ivanmichal** [comment]
OK, so here's what worked for me:
* I've installed the latest Tapo 3.8.113 via ApkMirror app (first install their app, then download the package from their website, then install the downloaded package via the app)
* Switched on the Third Party Integration in Tapo App
* (at this point, even after few hours, the integration was still not wroking)
* logged in to Tapo website: https://tapo.tplinkcloud.com/
* tried the integration Login again couple of minutes later and voila !

**Comment by hyperanalysis** [comment]
v3.8.115 available in Play Store for Android for me today! Downloaded, set the 'Lab' settings, rebooted HA, success! Profit! Rejoice! ❤️

**Reply by ivan11paths** [reply]
I also have the app available from the Play Store in Spain. Cameras configured without any problems.

**Comment by joe7045** [comment]
Argggg, I think I´m the only one that still have troubles with this issue.
3.8.115 installed
lab thirth party active
reset camera and tplink passwords
verified 2 steps disabled
cameras and ha rebooted several times and ... always the same

**Reply by JurajNyiri** [reply]
Factory reset, readd to force it as described in FAQ.

**Comment by joe7045** [comment]
It works!!!
Thank you very much. I was really close to throw my cameras to the bin.
Only thing now is that motion sensor didn´t appear, but after serveral restarts, one of them have it....only 4 remaining but I think it´s only time.
Thank you again
